src/core/agent/services/signifyNotificationService.ts

```diff
@@ -438,6 +438,19 @@ class SignifyNotificationService extends AgentService {
               });
               break;
             }
+            case OperationPendingRecordType.SubmitAdmitCredential: {
+              const credentialId = pendingOperation?.metadata?.credentialId;
+              if (credentialId) {
+                await Agent.agent.ipexCommunications.waitAndUpdateCredential(
+                  credentialId
+                );
+              }
+              callback({
+                opType: pendingOperation.recordType,
+                oid: recordId,
+              });
+              break;
+            }
```

Comment:
Does something like this work:

```
case OperationPendingRecordType.Exchange: {
  const exn = await this.props.signifyClient.exchanges().get(said);
  if (exn.r === NotificationRoute.ExnIpexAgree) {
    const grant = await this.props.signifyClient.exchanges().get(exn.p);
    const credentialId = grant.e.acdc.d;
    await Agent.agent.ipexCommunications.waitAndUpdateCredential(credentialId);
    callback({
      opType: pendingOperation.recordType,
      oid: recordId,
    });
    break;
  }
}
```

Reply:
The above would allow us to easily handle other types of exchange messages while keeping our storage light

Reply:
Ok, this works. I will make an update!